staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -800,7 +800,8 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericapifilters.WithCacheControl(handler)
 	handler = genericfilters.WithHSTS(handler, c.HSTSDirectives)
 	if c.ShutdownSendRetryAfter {
-		handler = genericfilters.WithRetryAfter(handler, c.lifecycleSignals.AfterShutdownDelayDuration.Signaled())
+		shouldRetryAfterFn := genericfilters.NewShouldRespondWithRetryAfterFunc(c.ShutdownSendRetryAfter, c.lifecycleSignals.AfterShutdownDelayDuration.Signaled())
```

Comment:
the NewShouldRespondWithRetryAfterFunc helper should be a prive func here turning the signal into a predicate `func() bool`.

Reply:
comment still applies. Make it a local helper. This is a trivial func part of the plumbing here.